lumen_runtime/src/otp/erlang/float_to_string.rs

```diff
@@ -253,10 +206,59 @@ impl TryFrom<Term> for OptionsBuilder {
 
                     continue;
                 }
-                _ => return Err(badarg!().into()),
+                _ => return Err(TryFromTermError::Type),
             }
         }
 
         Ok(options_builder)
     }
 }
+
+#[derive(Debug, Error)]
+pub enum TryFromTermError {
```

Comment:
Aren't we basically just surfacing this as a badarg? Why not just use `anyhow::Error` here as well, and use `.context(&format!("....", ...))` to specify what specifically went wrong?

As a general rule, we should only define error variants when we want to dispatch on them, otherwise we should just roll them up into a single type.